fenix/app/src/main/java/org/mozilla/fenix/browser/StandardSnackbarErrorBinding.kt

```diff
@@ -0,0 +1,97 @@
+
+    override suspend fun onState(flow: Flow<AppState>) {
+        flow.map { state -> state.standardSnackbarError }
+            .distinctUntilChanged()
+            .collect { standardSnackbarError ->
+                if (standardSnackbarError?.standardSnackbarErrorVisibility == true) {
+                    activity.getRootView()?.let { view ->
+                        val snackBar = FenixSnackbar.make(
+                            view = view,
+                            duration = Snackbar.LENGTH_INDEFINITE,
+                            isDisplayedWithBrowserToolbar = true,
+                        )
+                        snackBar.setText(
+                            standardSnackbarError.standardSnackbarErrorMessage
+                                ?: activity.getString(R.string.standard_snackbar_error_message),
+                        )
+                        snackBar.setButtonTextColor(
+                            ContextCompat.getColor(
+                                activity,
+                                R.color.fx_mobile_text_color_primary,
+                            ),
+                        )
+                        snackBar.setBackground(
+                            AppCompatResources.getDrawable(
+                                activity,
+                                R.drawable.standard_snackbar_error_background,
+                            ),
+                        )
+                        snackBar.setSnackBarTextColor(
+                            ContextCompat.getColor(
+                                activity,
+                                R.color.fx_mobile_text_color_warning,
+                            ),
+                        )
+                        snackBar.setAction(
+                            text = activity.getString(R.string.standard_snackbar_error_dismiss),
+                            action = {
+                                view.context.components.appStore.dispatch(
+                                    AppAction.ShowStandardSnackbarErrorAction(
+                                        StandardSnackbarError(
+                                            false,
+                                            activity.getString(
+                                                R.string.unable_to_save_to_pdf_error,
+                                            ),
+                                        ),
+                                    ),
+                                )
+                            },
+                        )
+                        snackBar.show()
+                    }
+                }
+            }
+    }
+}
+
+/**
+ * Standard Snackbar Error data class.
+ *
+ * @param standardSnackbarErrorMessage that will appear on the snackbar.
+ * @param standardSnackbarErrorVisibility of the snackbar.
+ */
+data class StandardSnackbarError(
+    val standardSnackbarErrorVisibility: Boolean = false,
+    val standardSnackbarErrorMessage: String? = null,
```

Comment:
I think `messsage` should be non-null, as not sure an use case where we could have a `SnackbarError` without a message, checking closely I'm not sure if we need `standardSnackbarErrorVisibility` could you help me to understand why do we need to include a `visibility` field ?

Reply:
I was imagining that we would show the `Snackbar` and after sometime it will go away, as topical Snackbars

Reply:
The snackbar will stay on the screen until the user presses the dismiss button. I show and hide the snack based on visibility flag.

Reply:
Thanks for the clarification, that's a bit odd behaviour for a `Snackbar`, is this a product/UI/UX suggestion?

Reply:
Perfect, thanks for the clarification!